File: API/Data/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace API.Data.Entities
{
    public class Order
    {
        [Key]
        public int Id { get; set; }

        public DateTime OrderDate { get; set; }

        public decimal Subtotal { get; set; }

        //Navigation Properties
        public IEnumerable<OrderItem> OrderItems { get; set; }
            = Enumerable.Empty<OrderItem>();

        public AppUser AppUser { get; set; }

        public int AppUserId { get; set; }
    }

}

File: clientel/src/app/_models/order.ts
export class OrderItem {
    quantity: number;
    price: number;
    orderId: number = 0;
    gameId: number;
    game: string;
    gamePhotoUrl: string;
}

export class Order {
    id: number = 0;
    appUserId: number;
    orderDate: Date;
    subtotal: number;
    orderItems: OrderItem[] = [];
}

